core/src/main/java/io/substrait/expression/doc-files/FunctionArg.md

```diff
@@ -0,0 +1,54 @@
+## Representing and mapping Function arguments
+- `FunctionArg` is a marker interface
+- subtypes are `Expression`, `Type`, and  the new `EnumArg` class
```

Comment:
What's an example of a FunctionArg that is a Type? Is that when the argument requires the name of a type? Can that be modeled as an Enum whose possible values must be a Type (i.e. the possible values don't need to be enumerated)?

Reply:
I don't have an example.
But `Argument` in `Function` has a notion of `TypeArgument`. I am assuming this is modeling generic function invocations.

Reply:
Not sure what this is either. Maybe @jinfengni  or @jacques-n know?